test/hotspot/jtreg/compiler/c2/TestDivModNodes.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.Test;
+
+import java.util.function.BiFunction;
+
+/*
+ * @test
+ * @summary Test DIV and MOD nodes are converted into DIVMOD where possible
+ * @requires os.arch=="amd64" | os.arch=="x86_64" | os.arch=="aarch64"
+ * @library /test/lib /
+ * @run main/othervm -XX:+UseDivMod compiler.c2.TestDivModNodes
+ * @run main/othervm -XX:-UseDivMod compiler.c2.TestDivModNodes
+ */
```

Comment:
Drive by comment: please allow it to run on any platform, and add restrictions on whee the IR rules are run. That allows us to test results on other platforms, which is valuable on its own.

Reply:
Done. Waiting GHA to complete